### PR TITLE
proxychains{,-ng}: swap priority 4 and 5 in get_config_path

### LINKED ISSUE #136093: proxychains not finding the correct config
### Describe the bug

Instead of reading config file `/etc/static/proxychains.conf` generated with `programs.proxychains`, proxychains reads config from the program's store path.

### Steps To Reproduce

1. Edit `configuration.nix`
   
   ```nix
   programs.proxychains = {
     enable = true;
     proxies = {
       myproxy = {
         enable = true;
         type = "http";
         host = "192.168.101.2";
         port = 10808;
       };
     };
   };
   ```
2. Run `nixos-rebuild switch`
3. Run `proxychains4 curl https://nixos.org` and check its output
   
   ```console
   [chuang@nixos:~]$ proxychains4 curl https://nixos.org
   [proxychains] config file found: /nix/store/pz9izfrayk2hymgygmqzwdgh481fz4pc-proxychains-4.4.0/etc/proxychains.conf
   [proxychains] preloading /nix/store/pz9izfrayk2hymgygmqzwdgh481fz4pc-proxychains-4.4.0/lib/libproxychains4.so
   [proxychains] DLL init
   [proxychains] Strict chain  ...  127.0.0.1:9050  ...  timeout
   curl: (7) Couldn't connect to server
   ```

   Here proxychains uses `127.0.0.1:9050` (from the default config) as proxy.

### Expected behavior

proxychains reads config from `/etc/static/proxychains.conf`, and uses `192.168.101.2:10808` as proxy.

### Notify maintainers

@sorki @viric @peti @fabaff @womfoo @Fuuzetsu @siraben 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[chuang@nixos:~]$ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/wv1mbc34jaws00lyhgqr6aix5ql0c1r6-bash-interactive-4.4-p23-dev
copying path '/nix/store/wv1mbc34jaws00lyhgqr6aix5ql0c1r6-bash-interactive-4.4-p23-dev' from 'https://mirrors.ustc.edu.cn/nix-channels/store'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.57, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos"`
 - channels(chuang): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- pkgs.proxychains
# a list of nixos modules affected by the problem
module:
- programs.proxychains
```

### PR DESCRIPTION
**Pull request by chuangzhu**
###### Description of changes

Prefer `/etc/proxychains.conf` to the one in Nix store. Fixes #136093.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sorki — APPROVED** [review]
LGTM. Thank you, would be cool to propose the patch upstream as well.

**Review by ZenithalHourlyRate — APPROVED** [review]
LGTM. This can build on my x86_64-linux machine.

> would be cool to propose the patch upstream as well.

Agreed.

File: pkgs/tools/networking/proxychains-ng/default.nix [hunk 1]
```diff
@@ -25,6 +25,7 @@ stdenv.mkDerivation rec {
       url = "https://github.com/rofl0r/proxychains-ng/commit/9b42da71f4df7b783cf07a58ffa095e293c43380.patch";
       sha256 = "sha256-tYv9XP51WtsjaoklwQk3D/MQceoOvtdMwBraECt6AXQ=";
     })
+    ./swap-priority-4-and-5-in-get_config_path.patch
```
Comment (bobby285271) [inline pkgs/tools/networking/proxychains-ng/default.nix:28]:
```suggestion
    # https://github.com/NixOS/nixpkgs/issues/136093
    ./swap-priority-4-and-5-in-get_config_path.patch
```

File: pkgs/tools/networking/proxychains/default.nix [hunk 2]
```diff
@@ -14,6 +14,8 @@ stdenv.mkDerivation rec {
     sha256 = "083xdg6fsn8c2ns93lvy794rixxq8va6jdf99w1z0xi4j7f1nyjw";
   };
 
+  patches = [ ./swap-priority-4-and-5-in-get_config_path.patch ];
```
Comment (bobby285271) [inline pkgs/tools/networking/proxychains/default.nix:17]:
```suggestion
  patches = [
    # https://github.com/NixOS/nixpkgs/issues/136093
    ./swap-priority-4-and-5-in-get_config_path.patch
  ];
```

**Comment by bobby285271** [comment]
@ofborg build proxychains proxychains-ng

**Pull request merged**